GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -17,20 +17,26 @@
 
 import static org.gearvrf.utility.Assert.*;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.gearvrf.utility.Exceptions;
+import org.gearvrf.utility.Log;
 
 /**
  * This is one of the key GVRF classes: It holds GL meshes.
  * 
  * A GL mesh is a net of triangles that define an object's surface geometry.
  */
-public class GVRMesh extends GVRHybridObject {
+public class GVRMesh extends GVRHybridObject implements PrettyPrint {
```

Comment:
Not questioning the usefulness at all - it is nice to have it. What if we had a separate pretty-printing class that does all the work, knows the nodes and performs the formatting in one place? Instead of having the interface and the code in all the nodes of interest.